libminifi/src/utils/file/FileUtils.cpp

```diff
@@ -74,13 +74,23 @@ time_t to_time_t(std::filesystem::file_time_type file_time) {
 #endif
 }
 
-std::chrono::time_point<std::chrono::system_clock> to_sys(std::filesystem::file_time_type file_time) {
-#if defined(WIN32) || defined(_LIBCPP_VERSION) && _LIBCPP_VERSION >= 14000
-  return std::chrono::time_point_cast<std::chrono::system_clock::duration>(file_time - std::filesystem::file_time_type::clock::now() + std::chrono::system_clock::now());
-#elif defined(_LIBCPP_VERSION)
+std::chrono::system_clock::time_point to_sys(std::filesystem::file_time_type file_time) {
+#if defined(WIN32)
+  return std::chrono::clock_cast<std::chrono::system_clock>(file_time);
+#elif defined(_LIBCPP_VERSION) && (_LIBCPP_VERSION < 14000)
   return std::chrono::system_clock::from_time_t(std::chrono::file_clock::to_time_t(file_time));
 #else
-  return std::chrono::file_clock::to_sys(file_time);
+  return std::chrono::time_point_cast<std::chrono::system_clock::duration>(std::chrono::file_clock::to_sys(file_time));
+#endif
+}
+
+std::filesystem::file_time_type from_sys(std::chrono::system_clock::time_point sys_time) {
+#if defined(WIN32)
+  return std::chrono::clock_cast<std::chrono::file_clock>(sys_time);
+#elif defined(_LIBCPP_VERSION) && (_LIBCPP_VERSION < 14000)
+  return std::chrono::file_clock::from_time_t(std::chrono::system_clock::to_time_t(sys_time));
+#else
+  return std::chrono::file_clock::from_sys(sys_time);
```

Comment:
For which compiler is the duration cast needed in `to_sys`?  Does it need to be added for that compiler here in `from_sys`, as well?

Reply:
On libc++ (if the _LIBCPP_HAS_NO_INT128 is not defined) file_clock uses a 128 bit resolution while the system_clock only uses 64 bit, due to this we only need the cast in from_sys.

Reply:
I don't get it.  If `std::chrono::file_clock::to_sys(file_time)` returns 128 bits and needs to be duration-cast to 64 bits, then doesn't `std::chrono::file_clock::from_sys(sys_time)` return 64 bits and needs to be duration-cast to 128 bits?

Reply:
Ah, got it.  The duration conversion happens automatically in the `return` statement, but we need the `time_point_cast` when converting from a 128-bit underlying type to a 64-bit one (but not the other way round).

That's a bit nasty -- I hope `clock_cast` will fix this problem when it's available on all compilers.

Reply:
You only need to cast it where it can lose precision otherwise the cast is implicit.
So from hours to minutes you dont need duration_cast while you would need in the other direction.

Of course we could add the explicit cast to other function aswell, if we expect that on some platfrom system_clock will be more precise than file_clock (or simply to improve readability) What do you think?

Reply:
Yes, I think adding the `time_point_cast` in the unnecessary direction would help to make the code less confusing to humans.  Hopefully, the compiler will optimize it out.  Even more hopefullier, we can soon remove all this mess and simply use `clock_cast`.

Reply:
It turns out we can't rely on clock_cast on windows either, because (for some reason it needs timezone information which should be available after msi install but not neccesarly there during tests of course we could load it like CronTests do, but I think we can skip the hassle since the other platforms dont have clock_cast)

Anyways, I've refactored these in https://github.com/apache/nifi-minifi-cpp/pull/1560/commits/d8e7f065500dc282786918880057f7a23ca0b942

I've also removed the to_time_t functions from FileUtils since it was only used in a single test (ListFileTests) https://github.com/apache/nifi-minifi-cpp/pull/1560/commits/256d695e88751dbf5519546f53eebb72e4f3776f